x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx

```diff
@@ -190,6 +190,10 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     threatIndex: threatIndicesConfig,
   };
 
+  if (initialState.index == null) {
+    initialState.index = [];
```

Comment:
I was playing around with this form a bit, and it looks like you can do away with this guard by adding

```js
defaultValue: [],
```

to the form schema for the `index` field 👍

Reply:
i.e. 

```diff
diff --git a/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx b/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx
index f19bd044b6e..5d70dfd1a4b 100644
--- a/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx
+++ b/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/index.tsx
@@ -190,10 +190,6 @@ const StepDefineRuleComponent: FC<StepDefineRuleProps> = ({
     threatIndex: threatIndicesConfig,
   };
 
-  if (initialState.index == null) {
-    initialState.index = [];
-  }
-
   const { form } = useForm<DefineStepRule>({
     defaultValue: initialState,
     options: { stripEmptyFields: false },
diff --git a/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx b/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx
index c863eb8cf1e..c6e0085ef77 100644
--- a/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx
+++ b/x-pack/plugins/security_solution/public/detections/components/rules/step_define_rule/schema.tsx
@@ -44,6 +44,7 @@ import {
 
 export const schema: FormSchema<DefineStepRule> = {
   index: {
+    defaultValue: [],
     fieldsToValidateOnChange: ['index', 'queryBar'],
     type: FIELD_TYPES.COMBO_BOX,
     label: i18n.translate(
```